### DISCUSSION TITLE
Webmin FirewallD module: Features request

### DISCUSSION THREAD

**Discussion by iliaross** [webmin/webmin#1754] [Ideas]
This discussion is dedicated to Webmin FirewallD module.

Any suggestions are welcome!

**Comment by chris001** [comment]
There was an issue related to `nftables` #1622 #423 also [issues in in virtualmin-gpl](https://github.com/virtualmin/virtualmin-gpl/issues?q=is%3Aissue+firewalld+is%3Aopen)

**Reply by iliaross** [reply]
Does FirewallD have issues with `nftables`_**?**_

**Reply by chris001** [reply]
I don't  believe it has problem with nftables, however many users were requesting webmin support using nftables on the system, not iptables.  Firewalld lets them support nftables but they wanted it nativelt?

**Comment by iliaross** [comment]
Those users wanted support for `nftables` in old Webmin Linux Firewall module. We use FirewallD API (with Webmin FirewallD module), and this is no longer our problem (whether system uses `nftables` or `legacy` iptables). It will just work.

**Comment by flappadap** [comment]
Zypper support in the Software Packages module would be nice for open-SuSE

**Comment by kupietools** [comment]
Sure would be nice if we could add sources to zones. For instance, if the "trusted" zone must be empty by default, it stands to reason there should be a way to add local addresses to it. 

Right now I've just lost about 5 hours troubleshooting because installing webmin, without any warning that I noticed, closed off almost all my server's services so I can no longer access them from the local network. If I could add the local network to the "trusted" zone, that would be great, but in several hours of searching I haven't found any way to do that, now it's 7AM and I have 25 wifi lights that I can't turn off for the night and go to bed because webmin firewalled my entire home network out of the home automation service I need to turn them off. I can't find any way to edit the zones, nor will it let me delete and recreate the trusted zone, so it's just sitting there, empty and uneditable in any way that I can find. 

Even searching Webmin's built-in search and every associated support website for "source" and "zone" for 4 hours hasn't turned up how I add anything to the Trusted zone. And my wifi lights are burning my eyeballs out.  

So either that ought to be fixed, or, there needs to be better or more visible documentation on how to do it.

**Reply by jcameron** [reply]
That sounds like a good feature request, but also Webmin doesn't adjust the firewall configuration at install time. And if we were to add a feature like this, it would be to open up port 10000, not close other ports down.  Unless you were using Webmin's FirewallD module after installing and it messed up the config somehow?

**Comment by JimTR** [comment]
The firewalld module is nowhere near as good as the iptables module which lays out the data in a logical way, unlike the firewalld module. Also firewalld & fail2ban don't  work that well together, when I hit the first issue of firewalld not being able to process the requests from fail2ban in a timely manner, I switched fail2ban to use iptables rather than firewaald, which did improve matters greatly. That said when you are in this way of using fail2ban IPV6 address were not banned to either iptables or firewalld. Net result was to dump both firewalld & fail2ban & code my own, which (to date) has cured all the issues I had
